In [ ]:
# Illustrating the z-test
# developed by Eve Marenghi for PUI in 2018

In [1]:
from statistics import mean 

## IDEA: 
## The new bus route is improving commute 

## NULL HYPOTHESIS: 
## The commute time is the same or longer with the new bus route as it was before  

## ALTERNATIVE HYPOTHESIS: 
## The commute time is shorter with the new bus route than it was before 

# _$H_0$_ : $Tnew - Told >= 0$
# _$H_1$_ : $Tnew - Told < 0$

## Significance level:  $\alpha=0.05$ 

In [8]:
# Parameters for Told: 
told_mean = 36
told_stdev = 6

In [17]:
# Uploaded times.txt to HW4_em3932 folder
# Read the data into a list 
with open("times.txt", "r") as tnew_file:
    tnew = tnew_file.read().splitlines()

In [6]:
# Convert the string elements to floats 
tnew = [float(i) for i in tnew]

In [7]:
# Calculate the mean and find the sample size  
tnew_mean = mean(tnew)
tnew_size = len(tnew)
print(tnew_mean)
print(tnew_size)

34.4661616883
100


## Formula: 

$Z = \frac{\mu_{pop} - \mu_{sample}}{\sigma / \sqrt{N}}$

In [15]:
# Calculate z 
z = (told_mean - tnew_mean) / (told_stdev / (tnew_size)**(1/2))
print(z)

2.55639718616667


## We set the significance level to $\alpha=0.05$. Since $2.56 >= 2$ we are further than 2 sigmas away from the old mean and we reject the null hypothesis. 